universal-application-tool-0.0.1/app/views/FileUploadStrategy.java

```diff
@@ -0,0 +1,192 @@
+import play.mvc.Http.HttpVerbs;
+import services.MessageKey;
+import services.applicant.question.ApplicantQuestion;
+import services.applicant.question.FileUploadQuestion;
+import views.applicant.ApplicantProgramBlockEditView.Params;
+import views.questiontypes.ApplicantQuestionRendererFactory;
+import views.questiontypes.ApplicantQuestionRendererParams;
+import views.questiontypes.FileUploadQuestionRenderer;
+import views.style.ApplicantStyles;
+import views.style.Styles;
+
+public abstract class FileUploadStrategy {
+
+  static final String IMAGES_AND_PDF = "image/*,.pdf";
+  final String BLOCK_FORM_ID = "cf-block-form";
+  private final String FILEUPLOAD_CONTINUE_FORM_ID = "cf-fileupload-continue-form";
+  private final String FILEUPLOAD_DELETE_FORM_ID = "cf-fileupload-delete-form";
+
+  public abstract ContainerTag signedFileUploadFields(
+      ApplicantQuestionRendererParams params, FileUploadQuestion fileUploadQuestion);
+
+  public abstract Tag renderFileUploadBlockSubmitForms(
+      Params params, ApplicantQuestionRendererFactory applicantQuestionRendererFactory);
+
+  /**
+   * Returns two hidden forms for navigating through a file upload block without uploading a file.
+   *
+   * <p>Delete form sends an update with an empty file key. An empty file key erases the existing
+   * file key if one is present. In either case, the file upload question is marked as seen but
+   * unanswered, namely skipping the file upload. This is only allowed for an optional question.
+   *
+   * <p>Continue form sends an update with the currently stored file key, the same behavior as an
+   * applicant re-submits a form without changing their answer. Continue form is only used when an
+   * existing file (and file key) is present.
+   */
+  Tag renderDeleteAndContinueFileUploadForms(Params params) {
+    String formAction =
+        routes.ApplicantProgramBlocksController.update(
+                params.applicantId(), params.programId(), params.block().getId(), params.inReview())
+            .url();
+    ApplicantQuestionRendererParams rendererParams =
+        ApplicantQuestionRendererParams.builder().setMessages(params.messages()).build();
+
+    Tag continueForm =
+        form()
+            .withId(FILEUPLOAD_CONTINUE_FORM_ID)
+            .withAction(formAction)
+            .withMethod(HttpVerbs.POST)
+            .with(makeCsrfTokenInputTag(params.request()))
+            .with(
+                each(
+                    params.block().getQuestions(),
+                    question -> renderFileKeyField(question, rendererParams)));
+    Tag deleteForm =
+        form()
+            .withId(FILEUPLOAD_DELETE_FORM_ID)
+            .withAction(formAction)
+            .withMethod(HttpVerbs.POST)
+            .with(makeCsrfTokenInputTag(params.request()))
+            .with(
+                each(
+                    params.block().getQuestions(),
+                    question -> renderEmptyFileKeyField(question, rendererParams)));
```

Comment:
question (blocking): is this and the `continueForm` an AWS-specific implementation, or will it work for Azure as well? As in, will passing in an empty file input delete the file in Azure Blob? If not, these should probably go in the AWS strategy file.

Reply:
I need to test further what will work for Azure, but I'l leave that for the next PR where I implement Azure file upload. Moving these functions to AWS strategy for now.